pulldown-cmark/src/tree.rs

```diff
@@ -186,8 +186,6 @@ impl<T: Default> Tree<T> {
         }
         if next.is_some() {
             self.cur = next;
-        } else {
-            self.pop();
```

Comment:
I have no idea what I was thinking when I wrote this. It doesn't make sense, and it doesn't seem necessary for anything.

Reply:
I though the same finding the bug. Looks to me like some sort of premature optimization.